From e2ba5a30e8888ffd29bf02ca867430c6371fbc4d Mon Sep 17 00:00:00 2001
From: "P. Sadik" <psadik@broadcom.com>
Date: Fri, 5 Jul 2013 11:10:41 +0530
Subject: [PATCH 0379/1532] initramfs: Make initramfs independent of Linux
 Kernel.

Instead, load initramfs image separately and specify the address
through dts files.
[Based on SDK 3.2]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 arch/mips/boot/dts/mn4xlp8xx-linux.dts    |  8 +++++---
 arch/mips/boot/dts/mnxlp8xx-linux.dts     |  6 ++++--
 arch/mips/boot/dts/xlp2xx-ep-linux.dts    |  6 ++++--
 arch/mips/boot/dts/xlp2xx-linux.dts       | 10 ++++++----
 arch/mips/boot/dts/xlp3xx-linux.dts       |  8 +++++---
 arch/mips/boot/dts/xlp3xx-xmc-linux.dts   |  6 ++++--
 arch/mips/boot/dts/xlp8xx-linux.dts       |  4 +++-
 arch/mips/boot/dts/xlp9xx-linux-guest.dts |  2 ++
 arch/mips/boot/dts/xlp9xx-linux.dts       |  2 ++
 9 files changed, 35 insertions(+), 17 deletions(-)

diff --git a/arch/mips/boot/dts/mn4xlp8xx-linux.dts b/arch/mips/boot/dts/mn4xlp8xx-linux.dts
index 5873acb..5209d02 100644
--- a/arch/mips/boot/dts/mn4xlp8xx-linux.dts
+++ b/arch/mips/boot/dts/mn4xlp8xx-linux.dts
@@ -11,11 +11,11 @@
 	/*memory range for linux. This will not appear in dom0*/
         memory {
                 device_type = "memory";
-                reg =  <0 0x01000000 0 0x0B000000   
+                reg =  <0 0x01000000 0 0x0B000000
                         0 0x20000000 0 0xa0000000
                         0 0xe0000000 0 0x20000000>;
         };
-	
+
 	hypervisor {
 		hypervisor-name = "Xen";
 		alloc_dom0_memory = <0>;
@@ -53,7 +53,7 @@
                                 node_0_vc_mask = <0xbbbbbbbb 0xbbbbbbbb 0xbbbbbbbb 0xbbbbbbbb>;
                                 node_1_vc_mask = <0xbbbbbbbb 0xbbbbbbbb 0xbbbbbbbb 0xbbbbbbbb>;
                                 node_2_vc_mask = <0xbbbbbbbb 0xbbbbbbbb 0xbbbbbbbb 0xbbbbbbbb>;
-                                node_3_vc_mask = <0xbbbbbbbb 0xbbbbbbbb 0xbbbbbbbb 0xbbbbbbbb>;                       
+                                node_3_vc_mask = <0xbbbbbbbb 0xbbbbbbbb 0xbbbbbbbb 0xbbbbbbbb>;
 			};
 			nae {
 				freein-fifo-mask = <0xffff 0xffff 0 0>;
@@ -76,6 +76,8 @@
 
 	chosen {
 		bootargs = " console=ttyS0,115200";
+		linux,initrd-start=<0x09000000>;
+		linux,initrd-end=<0x0b000000>;
 	};
 
 	/* These binaries are downloaded at the resp physical memory locations
diff --git a/arch/mips/boot/dts/mnxlp8xx-linux.dts b/arch/mips/boot/dts/mnxlp8xx-linux.dts
index 9e09038..e0d0080 100644
--- a/arch/mips/boot/dts/mnxlp8xx-linux.dts
+++ b/arch/mips/boot/dts/mnxlp8xx-linux.dts
@@ -11,7 +11,7 @@
 	/*memory range for linux. This will not appear in dom0*/
         memory {
                 device_type = "memory";
-                reg =  <0 0x01000000 0 0x0B000000   
+                reg =  <0 0x01000000 0 0x0B000000
                         0 0x20000000 0 0xa0000000
                         0 0xe0000000 0 0x20000000>;
         };
@@ -53,7 +53,7 @@
                                 node_0_vc_mask = <0xbbbbbbbb 0xbbbbbbbb 0xbbbbbbbb 0xbbbbbbbb>;
                                 node_1_vc_mask = <0xbbbbbbbb 0xbbbbbbbb 0xbbbbbbbb 0xbbbbbbbb>;
                                 node_2_vc_mask = <0xbbbbbbbb 0xbbbbbbbb 0xbbbbbbbb 0xbbbbbbbb>;
-                                node_3_vc_mask = <0xbbbbbbbb 0xbbbbbbbb 0xbbbbbbbb 0xbbbbbbbb>;                       
+                                node_3_vc_mask = <0xbbbbbbbb 0xbbbbbbbb 0xbbbbbbbb 0xbbbbbbbb>;
 			};
 			nae {
 				freein-fifo-mask = <0x3ffff 0x3ffff 0 0>;
@@ -76,6 +76,8 @@
 
 	chosen {
 		bootargs = " console=ttyS0,115200 ";
+		linux,initrd-start=<0x09000000>;
+		linux,initrd-end=<0x0b000000>;
 	};
 
 	/* These binaries are downloaded at the resp physical memory locations
diff --git a/arch/mips/boot/dts/xlp2xx-ep-linux.dts b/arch/mips/boot/dts/xlp2xx-ep-linux.dts
index 057c217..242926b 100644
--- a/arch/mips/boot/dts/xlp2xx-ep-linux.dts
+++ b/arch/mips/boot/dts/xlp2xx-ep-linux.dts
@@ -11,7 +11,7 @@
 	/*memory range for linux. This will not appear in dom0*/
         memory {
                 device_type = "memory";
-                reg =  <0 0x01000000 0 0x0B000000   
+                reg =  <0 0x01000000 0 0x0B000000
                         0 0x20000000 0 0xa0000000
                         0 0xe0000000 0 0x20000000>;
         };
@@ -41,7 +41,7 @@
 				sae-rx-sync-vc  = <3>; /* should be exclusive */
 				ipsec-async-vc  = <3>;
 				ipsec-sync-vc   = <2>;
-							  
+
 
 			};
 			uart {
@@ -79,6 +79,8 @@
 
 	chosen {
 		bootargs = " console=ttyS0,115200 rhash_entries=1048576 ";
+		linux,initrd-start=<0x09000000>;
+		linux,initrd-end=<0x0b000000>;
 	};
 
 	/* These binaries are downloaded at the resp physical memory locations
diff --git a/arch/mips/boot/dts/xlp2xx-linux.dts b/arch/mips/boot/dts/xlp2xx-linux.dts
index eb99b80..21d178e 100644
--- a/arch/mips/boot/dts/xlp2xx-linux.dts
+++ b/arch/mips/boot/dts/xlp2xx-linux.dts
@@ -4,7 +4,7 @@
 
 /dts-v1/;
 / {
-	
+
 	model = "netlogic,XLP-FVP";
         compatible = "netlogic,xlp";
         #address-cells = <2>;
@@ -12,7 +12,7 @@
 	/*memory range for linux. This will not appear in dom0*/
         memory {
                 device_type = "memory";
-                reg =  <0 0x01000000 0 0x0B000000   
+                reg =  <0 0x01000000 0 0x0B000000
                         0 0x20000000 0 0xa0000000
                         0 0xe0000000 0 0x20000000>;
         };
@@ -41,7 +41,7 @@
 				sae-rx-vc = <0>; /*  */
 				sae-rx-sync-vc  = <3>; /* should be exclusive */
 				ipsec-async-vc  = <3>;
-				ipsec-sync-vc   = <2>; 
+				ipsec-sync-vc   = <2>;
 
 			};
 			uart {
@@ -50,7 +50,7 @@
 				sharedcfg = <0x1f000000>;
 			};
                         fmn {
-				/* make the node mask to 0xffffffff for SRIO */  
+				/* make the node mask to 0xffffffff for SRIO */
                                 node_0_vc_mask = <0xbbbbbbbb 0xbbbbbbbb 0xbbbbbbbb 0xbbbbbbbb>;
                         };
 			nae {
@@ -75,6 +75,8 @@
 
 	chosen {
 		bootargs = " console=ttyS0,115200  rhash_entries=1048576";
+		linux,initrd-start=<0x09000000>;
+		linux,initrd-end=<0x0b000000>;
 	};
 
 	/* These binaries are downloaded at the resp physical memory locations
diff --git a/arch/mips/boot/dts/xlp3xx-linux.dts b/arch/mips/boot/dts/xlp3xx-linux.dts
index 82e0080..2bad3b2 100644
--- a/arch/mips/boot/dts/xlp3xx-linux.dts
+++ b/arch/mips/boot/dts/xlp3xx-linux.dts
@@ -11,7 +11,7 @@
 	/*memory range for linux. This will not appear in dom0*/
         memory {
                 device_type = "memory";
-                reg =  <0 0x01000000 0 0x0B000000   
+                reg =  <0 0x01000000 0 0x0B000000
                         0 0x20000000 0 0xa0000000
                         0 0xe0000000 0 0x20000000>;
         };
@@ -41,7 +41,7 @@
 				sae-rx-vc = <0>; /*  */
 				sae-rx-sync-vc  = <3>; /* should be exclusive */
 				ipsec-async-vc  = <3>;
-				ipsec-sync-vc   = <2>; /* should be exclusive either  
+				ipsec-sync-vc   = <2>; /* should be exclusive either
 							  ipsec-sync-vc  or soc/srio@node-0  */
 
 			};
@@ -54,7 +54,7 @@
                                 slots = <2>;
                         };
                         fmn {
-				/* make the node mask to 0xffffffff for SRIO */  
+				/* make the node mask to 0xffffffff for SRIO */
                                 node_0_vc_mask = <0xbbbbbbbb 0xbbbbbbbb 0xbbbbbbbb 0xbbbbbbbb>;
                                 node_1_vc_mask = <0xbbbbbbbb 0xbbbbbbbb 0xbbbbbbbb 0xbbbbbbbb>;
                                 node_2_vc_mask = <0xbbbbbbbb 0xbbbbbbbb 0xbbbbbbbb 0xbbbbbbbb>;
@@ -82,6 +82,8 @@
 
 	chosen {
 		bootargs = " console=ttyS0,115200 rhash_entries=1048576";
+		linux,initrd-start=<0x09000000>;
+		linux,initrd-end=<0x0b000000>;
 	};
 
 	/* These binaries are downloaded at the resp physical memory locations
diff --git a/arch/mips/boot/dts/xlp3xx-xmc-linux.dts b/arch/mips/boot/dts/xlp3xx-xmc-linux.dts
index c4c7fa4e..a6ab16e 100644
--- a/arch/mips/boot/dts/xlp3xx-xmc-linux.dts
+++ b/arch/mips/boot/dts/xlp3xx-xmc-linux.dts
@@ -11,7 +11,7 @@
 	/*memory range for linux. This will not appear in dom0*/
         memory {
                 device_type = "memory";
-                reg =  <0 0x01000000 0 0x0B000000   
+                reg =  <0 0x01000000 0 0x0B000000
                         0 0x20000000 0 0x28a00000>;
         };
 
@@ -76,7 +76,9 @@
 	};
 
 	chosen {
-		bootargs = " console=ttyS0,115200 initrd=2000000@09000000";
+		bootargs = " console=ttyS0,115200 rhash_entries=1048576 ";
+		linux,initrd-start=<0x09000000>;
+		linux,initrd-end=<0x0b000000>;
 	};
 
 	/* These binaries are downloaded at the resp physical memory locations
diff --git a/arch/mips/boot/dts/xlp8xx-linux.dts b/arch/mips/boot/dts/xlp8xx-linux.dts
index d0f06e7..54aab9d 100644
--- a/arch/mips/boot/dts/xlp8xx-linux.dts
+++ b/arch/mips/boot/dts/xlp8xx-linux.dts
@@ -11,7 +11,7 @@
 	/*memory range for linux. This will not appear in dom0*/
         memory {
                 device_type = "memory";
-                reg =  <0 0x01000000 0 0x0B000000   
+                reg =  <0 0x01000000 0 0x0B000000
                         0 0x20000000 0 0xa0000000
                         0 0xe0000000 0 0x20000000>;
         };
@@ -84,6 +84,8 @@
 	};
 	chosen {
 		bootargs = " console=ttyS0,115200  rhash_entries=1048576";
+		linux,initrd-start=<0x09000000>;
+		linux,initrd-end=<0x0b000000>;
 	};
 
 	/* These binaries are downloaded at the resp physical memory locations
diff --git a/arch/mips/boot/dts/xlp9xx-linux-guest.dts b/arch/mips/boot/dts/xlp9xx-linux-guest.dts
index 6f9869c..5e6eec1 100644
--- a/arch/mips/boot/dts/xlp9xx-linux-guest.dts
+++ b/arch/mips/boot/dts/xlp9xx-linux-guest.dts
@@ -43,6 +43,8 @@
 	chosen {
 		netlogic,cpumask = <0x1>;
 		bootargs = " console=ttyS0,115200  rhash_entries=1048576";
+		linux,initrd-start=<0x09000000>;
+		linux,initrd-end=<0x0b000000>;
 	};
 
 	soc {
diff --git a/arch/mips/boot/dts/xlp9xx-linux.dts b/arch/mips/boot/dts/xlp9xx-linux.dts
index 50cf9ab..c07eb83 100644
--- a/arch/mips/boot/dts/xlp9xx-linux.dts
+++ b/arch/mips/boot/dts/xlp9xx-linux.dts
@@ -94,6 +94,8 @@
 		 */
 		netlogic,cpumask = <0x01>;
 		bootargs = " console=ttyS0,115200  rhash_entries=1048576";
+		linux,initrd-start=<0x09000000>;
+		linux,initrd-end=<0x0b000000>;
 	};
 
 	/* These binaries are downloaded at the resp physical memory locations
-- 
1.9.1

